express-server/app.js

```diff
@@ -13,19 +13,26 @@ app.use(morgan("tiny"))
 app.use(cors())
 
 // fetches best recipes given user food items from API
-app.get('/apiuserrecipes/:ingredients', async (request, response) => {
-    const params = {
-        ingredients: request.params.ingredients,
-        number: 20,
+app.get('/apiuserrecipes/:ingredients/:diets/:allergies', async (request, response) => {
```

Comment:
nit: Usually for optional parameters like diets and allergies here, you would use URL Parameters.
 e.g. `/apiuserrecipes/:ingredients/?diets=vegan&allergies=dairy` so that they can be left off.